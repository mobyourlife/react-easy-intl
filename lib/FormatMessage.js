import React, { Component } from 'react';
import MessageFormat from 'messageformat';

class FormatMessage extends Component {
  constructor(props) {
    super(props);
    this.mf = new MessageFormat('en');
  }

  render() {
    if (!this.props.str) {
      return null;
    }

    const fn = this.mf.compile(this.props.str);
    const text = fn(this.props);
    return <span>{text}</span>
  }
}

export default FormatMessage;
