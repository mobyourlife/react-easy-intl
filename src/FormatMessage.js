import React, { Component } from 'react'
import MessageFormat from 'messageformat'
import Globalize from './Globalize'

export class FormatMessage extends Component {
  constructor (props) {
    super(props)
    this.mf = new MessageFormat(Globalize.getLocale())
  }

  render () {
    if (!this.props.children) {
      return null
    }

    const msg = Globalize.getMessage(this.props.children)
    const fn = this.mf.compile(msg)
    const text = fn(this.props)
    return <span>{text}</span>
  }
}

export default FormatMessage
