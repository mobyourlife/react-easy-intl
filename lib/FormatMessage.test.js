/* global it, expect */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { render } from 'enzyme'
import FormatMessage from './FormatMessage'

const HI_NAME = 'Hi {name}!'
const I_HAVE_N_BOOKS = 'I have {qtyBooks, plural, =0{no books} one{1 book} other{# books} }.'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<FormatMessage />, div)
})

it('renders expected text', () => {
  const component = renderer.create(
    <FormatMessage />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders passed string', () => {
  const wrapper = render(<FormatMessage str={HI_NAME} name='John Doe' />)
  expect(wrapper.text()).toEqual('Hi John Doe!')
})

it('pluralizes zero', () => {
  const wrapper = render(<FormatMessage str={I_HAVE_N_BOOKS} qtyBooks={0} />)
  expect(wrapper.text()).toEqual('I have no books.')
})

it('pluralizes one', () => {
  const wrapper = render(<FormatMessage str={I_HAVE_N_BOOKS} qtyBooks={1} />)
  expect(wrapper.text()).toEqual('I have 1 book.')
})

it('pluralizes number', () => {
  const wrapper = render(<FormatMessage str={I_HAVE_N_BOOKS} qtyBooks={100} />)
  expect(wrapper.text()).toEqual('I have 100 books.')
})
