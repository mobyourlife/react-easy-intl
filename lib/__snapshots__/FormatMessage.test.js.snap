exports[`test renders expected text 1`] = `null`;
